Sources/Fog/FogSyncChecker.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+import Foundation
+
+protocol FogSyncCheckable {
+    var viewsHighestKnownBlock: UInt64 { get }
+    var ledgersHighestKnownBlock: UInt64 { get }
+    var consensusHighestKnownBlock: UInt64 { get }
+    var currentBlockIndex: UInt64 { get }
+    var fogSyncThreshold: UInt64 { get }
```

Comment:
Just a suggestion... threshold doesn't always clearly indicate whether equality is acceptable.  I still catch myself using it in names too, but thinking something more semantic would be better, like 'maxAllowedBlockDelta', which clearly indicated equality via 'allowed', and delta indicating the difference between blocks.

Reply:
Good point about the naming, lets think of a better way to be more descriptive of the delta/threshold allowed. 

As for setting to 0, I don't think thats a major concern because this data structure isnt exposed outside of the SDK module, and not really intended for general consumption (tampering) by them, but maybe Im missing something with the access controls that could allow such a case.

Reply:
updated to `maxAllowedDelta`, much better now